core-bundle/src/Resources/contao/modules/ModuleProxy.php

```diff
@@ -19,20 +20,60 @@
  */
 class ModuleProxy extends Module
 {
+	/**
+	 * @var FrontendModuleReference
+	 */
+	private $reference;
+
+	/**
+	 * @param ModuleModel|Collection $objElement
+	 */
+	public function __construct($objElement, $strColumn = 'main')
+	{
+		if ($objElement instanceof Collection)
+		{
+			$objElement = $objElement->current();
+		}
+
+		if (!$objElement instanceof ModuleModel)
+		{
+			throw new \RuntimeException('ModuleProxy must be constructed with a ModuleModel');
+		}
+
+		$this->reference = new FrontendModuleReference($objElement, $strColumn);
+		$this->strColumn = $strColumn;
+
+		// Do not call parent constructor
+	}
+
 	/**
 	 * {@inheritdoc}
 	 */
 	public function generate()
 	{
-		$reference = new FrontendModuleReference($this->objModel, $this->strColumn);
 		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
 
 		if ($request && System::getContainer()->get('contao.routing.scope_matcher')->isBackendRequest($request))
 		{
-			$reference->setBackendScope();
+			$this->reference->setBackendScope();
 		}
 
-		return System::getContainer()->get('fragment.handler')->render($reference);
+		return System::getContainer()->get('fragment.handler')->render($this->reference);
+	}
+
+	public function __set($strKey, $varValue)
+	{
+		$this->reference->attributes['templateProps'][$strKey] = $varValue;
+	}
+
+	public function __get($strKey)
+	{
+		return $this->reference->attributes[$strKey]['templateProps'];
```

Comment:
Should this not be `->attributes['templateProps'][$strKey]` like above and below?

Reply:
Good catch! Fixed in f4209135eeb5b2d2b742d0b11f508201ea4af652